src/bucket/BucketSnapshotManager.h

```diff
@@ -45,28 +46,31 @@ class BucketSnapshotManager : NonMovableOrCopyable
   private:
     AppConnector mAppConnector;
 
+    // Lock must be held when accessing any member variables holding snapshots
+    mutable SharedMutex mSnapshotMutex;
+
     // Snapshot that is maintained and periodically updated by BucketManager on
     // the main thread. When background threads need to generate or refresh a
     // snapshot, they will copy this snapshot.
-    SnapshotPtrT<LiveBucket> mCurrLiveSnapshot{};
-    SnapshotPtrT<HotArchiveBucket> mCurrHotArchiveSnapshot{};
+    SnapshotPtrT<LiveBucket> mCurrLiveSnapshot GUARDED_BY(mSnapshotMutex){};
```

Comment:
Just curious, does this annotation support being on the left side of variable names? To me this reads like type information, and that would conform to our style a bit more.

Reply:
While this is supported, I followed the convention defined in the docs (https://clang.llvm.org/docs/ThreadSafetyAnalysis.html). I'd prefer to stay consistent with the docs to avoid confusion.